### ISSUE TITLE
Exception has occurred.

### ISSUE THREAD

**Issue by donjordano** [juliansteenbakker/mobile_scanner#806]
Trying the simple example on Android and the result is:
`PlatformException (PlatformException(MobileScanner, Called state before initializing., null, null))`

```
E/flutter (24413): [ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: type 'PlatformException' is not a subtype of type 'MobileScannerException' in type cast
E/flutter (24413): #0      _MobileScannerState._startScanner.<anonymous closure>.<anonymous closure> (package:mobile_scanner/src/mobile_scanner.dart:142:35)
E/flutter (24413): #1      State.setState (package:flutter/src/widgets/framework.dart:1138:30)
E/flutter (24413): #2      _MobileScannerState._startScanner.<anonymous closure> (package:mobile_scanner/src/mobile_scanner.dart:141:9)
E/flutter (24413): #3      _RootZone.runUnary (dart:async/zone.dart:1661:54)
E/flutter (24413): #4      _FutureListener.handleError (dart:async/future_impl.dart:174:22)
E/flutter (24413): #5      Future._propagateToListeners.handleError (dart:async/future_impl.dart:852:47)
E/flutter (24413): #6      Future._propagateToListeners (dart:async/future_impl.dart:873:13)
E/flutter (24413): #7      Future._completeError (dart:async/future_impl.dart:649:5)
E/flutter (24413): #8      _RootZone.runBinary (dart:async/zone.dart:1666:54)
E/flutter (24413): #9      _FutureListener.handleError (dart:async/future_impl.dart:171:22)
E/flutter (24413): #10     Future._propagateToListeners.handleError (dart:async/future_impl.dart:852:47)
E/flutter (24413): #11     Future._propagateToListeners (dart:async/future_impl.dart:873:13)
E/flutter (24413): #12     Future._completeError (dart:async/future_impl.dart:649:5)
E/flutter (24413): <asynchronous suspension>
E/flutter (24413):
```

**Comment by navaronbracke** [comment]
I'll take a look. Seems like a cast isn't working as intended.

**Comment by navaronbracke** [comment]
@donjordano Okay, so I see the issue.

There is a check that requires the plugin to be initialized in the _plugin binding_ before calling any other methods.
See https://github.com/juliansteenbakker/mobile_scanner/blob/master/android/src/main/kotlin/dev/steenbakker/mobile_scanner/MobileScannerHandler.kt#L100-L103

This initialization happens when setting up the MethodChannel bindings:

https://github.com/juliansteenbakker/mobile_scanner/blob/master/android/src/main/kotlin/dev/steenbakker/mobile_scanner/MobileScannerPlugin.kt#L21-L34

and

https://github.com/juliansteenbakker/mobile_scanner/blob/master/android/src/main/kotlin/dev/steenbakker/mobile_scanner/MobileScannerHandler.kt#L100-L103

Are you calling the `state` method very early? I.e. before `runApp()` ? You need to do `WidgetsFlutterBinding.ensureInitialized();`if you want to use any plugins before `runApp()` is executed in `void main() {}`.

See https://api.flutter.dev/flutter/widgets/WidgetsFlutterBinding/ensureInitialized.html

However, the `as MobileScannerException` cast failing is a bug, though.

**Comment by donjordano** [comment]
@navaronbracke thanks for the answer, I saw this was reported a while ago, but no answer there. 
In my case I use Riverpod. Nothing is called before runApp. Here is how it looks:

```
void main() async {
  runApp(
    const ProviderScope(child: FtsApp())
  );
}
```

**Comment by navaronbracke** [comment]
@donjordano Are you using a custom Flutter Engine or Android Activity perhaps?

Nevertheless, I updated the failing cast to also handle PlatformExceptions.
And I also think I found the reason why the cast was failing in the first place.

This fix (at least I hope it fixes it for you) is included in version 3.5.0
This version is being published to pub.dev at this very moment.

**Comment by donjordano** [comment]
Confirm. FIxed. Thanks a lot.
